lib/OpenQA/Test/TimeLimit.pm

```diff
@@ -21,6 +21,8 @@ my $SCALE_FACTOR = 1;
 sub import {
     my ($package, $limit) = @_;
     die "$package: Need argument on import, e.g. use: use OpenQA::Test::TimeLimit '42';" unless $limit;
+    # disable timeout if requested by ENV variable or running within debugger
+    return if $ENV{OPENQA_TEST_TIMEOUT_DISABLE} or defined $DB::single;
```

Comment:
looked into this together with @perlpunk and it looks like removing the `defined` should work. Seems like `$DB::single` is set to 1 while the debugger is running and not waiting on a command line prompt. So

```suggestion
    return if $ENV{OPENQA_TEST_TIMEOUT_DISABLE} or $DB::single;
```

Reply:
another fix is necessary.
`or` has lower precedence, so that means:
```
    (return if $ENV{OPENQA_TEST_TIMEOUT_DISABLE}) or $DB::single;
```
```suggestion
    return if ($ENV{OPENQA_TEST_TIMEOUT_DISABLE} or $DB::single);
```

Reply:
this also does not work. I tested with `perl -Ilib t/config.t ; make coverage KEEP_DB=1 TESTS=t/config.t; perl -Ilib -d t/config.t` and a custom code to die in lib/OpenQA/Test/TimeLimit.pm like:

```patch
-    return if $ENV{OPENQA_TEST_TIMEOUT_DISABLE} or defined $DB::single;
+    return if ($ENV{OPENQA_TEST_TIMEOUT_DISABLE} || (defined $DB::single && $DB::single == 0));
+        die "should reach here";
```

which should run into the die-call into the first two cases but not the third. But in case coverage analysis this does not trigger, meaning that the function did return early even though no interactive debugger was active, only Devel::Cover was used. I give up.

Reply:
try this:
```perl
return if ($ENV{OPENQA_TEST_TIMEOUT_DISABLE} or $INC{'perl5db.pl'});
```

Reply:
\o/ @perlpunk you are the best! That seems to work :)

Reply:
couldn't reopen so please see https://github.com/os-autoinst/os-autoinst-common/pull/7